Updated .travis config for php 5.6 and 7.4 and cakephp/app 3.8

othercorey opened the pull request:
I'm not sure how the original tests were passing when the app config was copied only for version 3.1.

This has a number of php notices in phpunit tests and deprecation warnings.

ADmad commented on tests/Fixture/composer.json:
```diff
@@ -11,7 +11,7 @@
     "require": {
-        "php": ">=5.4.16",
+        "php": ">=5.6",
         "cakephp/cakephp": "~3.0",
```
Cake version also needs to be updated. I don't remember off hand in which minor we bumped up to PHP 5.6.

othercorey commented:
Fixed the default cake config which fixed the php notices.

othercorey commented:
Fixed cakephp deprecation warnings except for `sendHeaders()`.  I don't know why it's called here.

```
        $response->sendHeaders();
        return new BrowserKitResponse(
            $response->getBody(),
            $response->getStatusCode(),
            $response->getHeaders()
        );
```

ADmad commented:
Could you please update the travis matrix to have a job which runs composer install with `--prefer-lowest` similar to https://github.com/cakephp/cakephp/blob/master/.travis.yml#L94. That will help ensure the updated methods exist in 3.4.

othercorey commented:
With the http deprecation fixes, this requires cakephp 3.6.  I updated the configs for this.

The --prefer-lowest solution doesn't seem to work with the way this script works.  I added a job that pinned the cakephp/app version which pins the cakephp version.

othercorey commented:
@dereuromark When you have some time, can you review this for us?

dereuromark commented on phpunit.xml.dist:
```diff
@@ -5,6 +5,7 @@
   stopOnFailure="false"
   syntaxCheck="false"
```
These should all be removed IMO.

dereuromark commented:
> @dereuromark When you have some time, can you review this for us?

what do you mean? in general, the changes look good.

othercorey commented:
> what do you mean? in general, the changes look good.

Yes please. I did this just to give cake some love, but I'm unfamiliar with the project still.  Admad wanted someone to review and you're so very very helpful ;)

dereuromark commented:
I am not a codecept guru though :) So I can only look on the high level, which looks fine.

dereuromark commented on src/Connector.php:
```diff
@@ -214,12 +177,12 @@ protected function handleError($exception)
     public function controllerSpy(Event $event)
     {
-        if (empty($event->data['controller'])) {
+        if (empty($event->getData('controller'))) {
```
empty() on method call? if (!...) suffices usually.

dereuromark commented on src/Helper/DbTrait.php:
```diff
@@ -71,8 +71,8 @@ public function haveRecord($model, $data = [])
-        $this->insertedRecords[$model][$table->primaryKey() . ' IN'][] = $data->{$table->primaryKey()};
-        return $data->{$table->primaryKey()};
+        $this->insertedRecords[$model][$table->getPrimaryKey() . ' IN'][] = $data->{$table->getPrimaryKey()};
+        return $data->{$table->getPrimaryKey()};
```
getPrimaryKey() can return array, I hate the way it can `string|array`... 
Maybe we assign this to a var first and throw RuntimeException if array?
